bpf/bpf_sock.c

```diff
@@ -213,8 +236,8 @@ static inline void sock4_handle_node_port(struct bpf_sock_addr *ctx,
 #endif /* ENABLE_NODEPORT */
 }
 
-__section("from-sock4")
-int sock4_xlate(struct bpf_sock_addr *ctx)
+static __always_inline int __sock4_xlate(struct bpf_sock_addr *ctx,
```

Comment:
Not for this PR, but just a thought since I had a similar problem in the `post_bind` hook with fake contexts: I wonder if it would make sense to introduce a new 3-tuple-type (i.e. `(ip_addr, port, proto)`) instead of passing fake contexts when unifying the lookup logic.

Reply:
You mean to change the service maps to have them unified as one?

Reply:
Ah, no - I was thinking about adding a new type for the fake context (so that this function signature would become a bit more obvious with regards to which ctx is fake and which is real). But nevermind, I just realized that this would not help in the example I hand in mind anyway.